From: Anton Khirnov <anton@khirnov.net>
Date: Wed, 20 Apr 2011 15:21:39 +0000 (+0200)
Subject: lavc: provide deprecated avcodec_thread_init until next major version
X-Git-Url: http://git.libav.org/?p=libav.git;a=commitdiff_plain;h=65af48b55930abe5ac9130be3ff4d9c287a7f010

lavc: provide deprecated avcodec_thread_init until next major version

It was deprecated only recently.
---

--- a/libavcodec/utils.c
+++ b/libavcodec/utils.c
@@ -1287,7 +1287,7 @@ void ff_thread_await_progress(AVFrame *f
 
 #endif
 
-#if LIBAVCODEC_VERSION_MAJOR < 53
+#if FF_API_THREAD_INIT
 
 int avcodec_thread_init(AVCodecContext *s, int thread_count)
 {
--- a/libavcodec/version.h
+++ b/libavcodec/version.h
@@ -86,5 +86,8 @@
 #ifndef FF_API_REQUEST_CHANNELS
 #define FF_API_REQUEST_CHANNELS (LIBAVCODEC_VERSION_MAJOR < 54)
 #endif
+#ifndef FF_API_THREAD_INIT
+#define FF_API_THREAD_INIT      (LIBAVCODEC_VERSION_MAJOR < 54)
+#endif
 
 #endif /* AVCODEC_VERSION_H */
--- a/libavcodec/avcodec.h
+++ b/libavcodec/avcodec.h
@@ -3684,7 +3684,7 @@ int avcodec_check_dimensions(void *av_lo
 
 enum PixelFormat avcodec_default_get_format(struct AVCodecContext *s, const enum PixelFormat * fmt);
 
-#if LIBAVCODEC_VERSION_MAJOR < 53
+#if FF_API_THREAD_INIT
 /**
  * @deprecated Set s->thread_count before calling avcodec_open() instead of calling this.
  */
